hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/security/DelegationTokenRenewer.java

```diff
@@ -1017,12 +1017,26 @@ public void run() {
                     "Exhausted max retry attempts {} in token renewer "
                         + "thread for {}",
                     tokenRenewerThreadRetryMaxAttempts, evt.getApplicationId());
+                continue;
               }
             }
           } catch (Exception e) {
             LOG.info("Problem in submitting renew tasks in token renewer "
                 + "thread.", e);
+            continue;
           }
+
+          // For normally finished DT renew event, it should be removed from futures.
+          futures.remove(evt);
+        }
+
+        // If the cluster is idle for some time, futures map is empty or no event handler found which may still cause high CPU utilization
+        // Therefore a short nap should be added here.
+        try {
+          Thread.sleep(1000);
```

Comment:
Set sleep 1000ms, what is the effect of this?

Reply:
Giving up CPU,

Reply:
> Set sleep 1000ms, what is the effect of this?

Give up CPU

Reply:
Can we give data? The CPU ratio before and after the modification, because it is given 1000ms, there seems to be no data support for this.

Reply:
@slfan1989 
Thanks for review. 
The CPU utilization statistics is as follows:
Before optimize:
![image](https://user-images.githubusercontent.com/78128883/209040946-9e954810-f7c1-4c07-ab3d-bae89104f4a8.png)

After optimize:
![image](https://user-images.githubusercontent.com/78128883/209040969-43dc5a32-5419-47ad-9be7-44691d71bbbb.png)

Reply:
This does not seem to be a good way to deal with it through thread sleep.

Reply:
Not convinced with the sleep either as @slfan1989 mentioned and it looks very arbitrary to me, worked for you but might cause issues for others